types/next-auth/client.d.ts

```diff
@@ -0,0 +1,54 @@
+import { FC } from 'react';
+import { NextPageContext } from 'next';
+
+interface Session {
+    user: {
+        name: string;
+        email: string;
+        image: string;
+    };
+    accessToken: string;
+    expires: string;
+}
+
+interface GetProvidersResponse {
+    [provider: string]: SessionProvider;
+}
+
+interface SessionProvider {
+    id: string;
+    name: string;
+    type: string;
+    signinUrl: string;
+    callbackUrl: string;
+}
+
+interface GenericObject {
+    [key: string]: any;
+}
+
+interface ContextProviderProps {
+    session: Session;
+    options?: ContextProviderOptions;
+}
+
+interface ContextProviderOptions {
+    site?: string;
+    basePath?: string;
+    clientMaxAge?: number;
+}
+
+type ContextProvider = FC<ContextProviderProps>;
+
+declare function useSession(): [Session, boolean];
+declare function getSession(context: NextPageContext): Promise<Session | null>;
+declare function session(context: NextPageContext): Promise<Session | null>;
+declare function getProviders(context: NextPageContext): Promise<GetProvidersResponse | null>;
+declare function providers(context: NextPageContext): Promise<GetProvidersResponse | null>;
+declare function getCsrfToken(context: NextPageContext): Promise<string | null>;
+declare function csrfToken(context: NextPageContext): Promise<string | null>;
+declare function signin(provider: SessionProvider, data: GenericObject): Promise<void>;
+declare function signout(context: NextPageContext): Promise<void>;
+declare const Provider: ContextProvider;
+
+export { useSession, getSession, session, getProviders, providers, getCsrfToken, csrfToken, signin, signout, Provider };
```

Comment:
I think this is incorrect. The actual export is an object exported as default, from what I understand, this is exporting each of the functions directly.

https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js#L323-L343

Reply:
Good catch 🧐 💯 , as you mention it's a _default export_ with the functions on it, I'll amend 👍🏻

Reply:
> I think this is incorrect. The actual export is an object exported as default, from what I understand, this is exporting each of the functions directly.

Its actually not, unfortunately. According to unpkg, [client.js](https://unpkg.com/browse/next-auth@3.1.0/client.js) looks like this:

```js
module.exports = require('./dist/client').default
```

So you end up with the module itself actually being `{ useSession, getSession, ... }` instead of a default export.

Reply:
@rbuckton, you're right. I was looking at `client/index.js` instead of `client.js` @lluia, my bad, could you switch it back?

Reply:
Yup I'll update 👍